Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp

```diff
@@ -1075,6 +1099,23 @@ JS_DEFINE_NATIVE_FUNCTION(RegExpPrototype::compile)
         if (!flags.is_undefined())
             return vm.throw_completion<TypeError>(ErrorType::NotUndefined, flags.to_string_without_side_effects());
 
+        // https://github.com/tc39/proposal-regexp-legacy-features#regexpprototypecompile--pattern-flags-
+        // 3. Let thisRealm be the current Realm Record.
+        auto* this_realm = vm.current_realm();
+
+        // 4. Let oRealm be the value of O’s [[Realm]] internal slot.
+        auto* o_realm = regexp_object->vm().current_realm();
```

Comment:
`regexp_object_realm` is better as we call `O` `regexp_object` - but needs the `[[Realm]]` slot, not just using the current realm